android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java

```diff
@@ -463,28 +464,49 @@ private void resetSwapFromToAssets() {
             if (swapFromAssetSymbol == null
                     || swapFromAssetSymbol.equals(eth.symbol)) { // default swap from ETH
                 updateBuySendSwapAsset(eth.symbol, eth, true);
+                resetSwapToAsset(eth, swapToAsset, swapFromAssetSymbol);
             } else {
                 mBlockchainRegistry.getTokenBySymbol(BraveWalletConstants.MAINNET_CHAIN_ID,
                         CoinType.ETH, swapFromAssetSymbol, token -> {
                             if (token != null) {
                                 updateBuySendSwapAsset(token.symbol, token, true);
+                                resetSwapToAsset(eth, swapToAsset, swapFromAssetSymbol);
+
+                                return;
                             }
+                            // We most likely have a custom token
+                            TokenUtils.getAllTokensFiltered(mBraveWalletService,
+                                    mBlockchainRegistry, mCurrentChainId, tokens -> {
+                                        for (BlockchainToken filteredToken : tokens) {
+                                            if (swapFromAssetSymbol.equals(filteredToken.symbol)) {
+                                                updateBuySendSwapAsset(
+                                                        filteredToken.symbol, filteredToken, true);
+                                                resetSwapToAsset(
+                                                        eth, swapToAsset, swapFromAssetSymbol);
+                                                break;
+                                            }
+                                        }
+                                    });
                         });
             }
+        }
+    }
 
-            if (mActivityType == ActivityType.SWAP) {
-                // Swap to
-                if (swapToAsset.equals(swapFromAssetSymbol)) { // swap from BAT
-                    updateBuySendSwapAsset(eth.symbol, eth, false);
-                } else {
-                    mBlockchainRegistry.getTokenBySymbol(BraveWalletConstants.MAINNET_CHAIN_ID,
-                            CoinType.ETH, swapToAsset, token -> {
-                                if (token != null) {
-                                    updateBuySendSwapAsset(token.symbol, token, false);
-                                }
-                            });
-                }
-            }
+    private void resetSwapToAsset(
+            BlockchainToken eth, String swapToAsset, String swapFromAssetSymbol) {
```

Comment:
`swapToAsset` doesn't need to be passed, it is always `"BAT"`.

Reply:
sounds good, it makes sense, I corrected it.